### DISCUSSION TITLE
Decimation equivalent?

### DISCUSSION THREAD

**Discussion by Carko** [awawa-dev/HyperHDR#424] [Q&A]
Hi!

When switching from Hyperion to HyperHDR I noticed really great/smooth performance, however one thing is "bothering" me - the resolution is too high. I have 101 LEDs so even small objects at the edge of the picture can cause a noticable change in LED. When there's a lot of motion, the LEDs are sometimes lighting up like there's a christmas tree behind the TV 😁- it looks very nice and accurate, but can be really distracting.

Increasing the "time" parameter in Smoothing obviously gets rid of this, but that also adds a noticeable delay and also filters out any significant but fast changes.

In Hyperion there's a "Decimation" parameter for the USB grabber that when set to 1/8 reduces the input resolution that effectively causes smoothing, but without additional delay. I tried the "quarter of frame mode" which does help, but not as much as "8" decimation.

I took another look at the Philips' Ambilight and it seems by default they are in fact doing more smoothing (besides having fewer LEDs than me).

Is there any way to achieve this, or am I maybe going completely in the wrong way?

![image](https://user-images.githubusercontent.com/3161616/203397411-02a2c735-3df4-4695-9c06-0c8d6c262951.png)

**Comment by awawa-dev** [comment]
Hi
"quarter of frame mode" reduces video frame width/height by 2 (final frame is 1/4 of the origin). Furthermore you can enable 'sparse processing' in the calibration tab to reduce the video frame processing width/height again by 2. So final area to process is 1/16 of the origin when quarter of frame & sparse processing are enabled. I think you want also to test anti-flickering filter in the smoothing processing (threshold 2 and with timeout enabled e.g. to 500) or maybe reduce framerate (frame rate decimation > 1) because I suspect that what you experience is probably not related to the frame resolution but rather to higher video processing speed.  You can also increase LED area size in the LED strip designer (to make them overlap a bit) so the visible resolution will be lower, without sharp transitions ... although 101 leds is a very small setup (typical setup has 250-350 these days).

**Comment by awawa-dev** [comment]
> (threshold 2 and with timeout enabled e.g. to 500)

because you are using analog grabber which has a high noise you may want to increase threshold to some higher value. 2 works fine with digital grabbers.

**Reply by Carko** [reply]
Thanks, I'll try both suggestions. I don't think grabber noise is what I am seeing, though.

In the meantime I added one drawing of what I mean to the top post - the top one is current situation, the bottom one is desired one.

I also thought that I have a small setup until I saw that Philips has much fewer LEDs than I - for example only 9 on each side on the latest model.

**Reply by awawa-dev** [reply]
Philips LED are different type: much stronger and higher quality (two different white channels for certain setups, wide angle) and can't be compared to typical tiny LEDs like ws2812x or even sk6812 RGBW. But their resolution is lower.

> In the meantime I added one drawing of what I mean to the top post - the top one is current situation, the bottom one is desired one.

I think that in the LED strip designer you should make the LEDs overlapped, so that the shape affects not only the directly adjacent LEDs, but also to some extent also the neighboring ones.

**Reply by awawa-dev** [reply]
Also in the processing you may test "Multicolor" (which is Hyperion like processing) instead of "Advanced squared" (better averaging for mix of colors, colors/luminescence is usually a bit stronger)

**Reply by Carko** [reply]
Thanks!!

The overlapping made a big difference and it's much smoother now, so for now I'll stick with it and try it a bit. I'll experiment with other settings as well.

Thanks, you've been a great help. Keep it up!
